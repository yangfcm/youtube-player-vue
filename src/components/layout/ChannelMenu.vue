<template>
  <div class="ui three item menu red stackable">
    <router-link class="item" to="videos" active-class="active">
      <i class="video icon"></i> Videos
    </router-link>
    <router-link class="item" to="playlist" active-class="active">
      <i class="list icon"></i>Playlist
    </router-link>
    <router-link class="item" to="intro" active-class="active">
      <i class="info circle icon"></i>Intro
    </router-link>
  </div>
</template>

<script>
export default {};
</script>

<style scoped>
</style>