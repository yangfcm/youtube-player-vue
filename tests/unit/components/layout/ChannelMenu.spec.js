import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import ChannelMenu from "@/components/layout/ChannelMenu.vue";

describe("Test ChannelMenu component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ChannelMenu, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("ChannelMenu component matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("component should render three links", () => {
    const links = wrapper.findAll("a");
    expect(links.length).toBe(3);
  });
});
