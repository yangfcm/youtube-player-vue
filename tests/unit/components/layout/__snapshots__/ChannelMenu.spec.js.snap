// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChannelMenu component ChannelMenu component matches snapshot 1`] = `
<div class="ui three item menu red stackable"><a class="item"><i class="video icon"></i> Videos
  </a> <a class="item"><i class="list icon"></i>Playlist
  </a> <a class="item"><i class="info circle icon"></i>Intro
  </a></div>
`;
